crates/hir_def/src/generics.rs

```diff
@@ -72,7 +72,12 @@ pub enum WherePredicateTypeTarget {
     // FIXME: ForLifetime(Vec<LifetimeParamId>, TypeRef)
 }
 
-type SourceMap = ArenaMap<LocalTypeParamId, Either<ast::Trait, ast::TypeParam>>;
+#[derive(Default)]
+pub struct SourceMaps {
+    // FIXME this shouldn't be public
```

Comment:
See Fixme below

Reply:
keeping stuff below the facede `hir` crate public is generally OK.